tests/test_converter.py

```diff
@@ -373,8 +373,12 @@ def fail(hint: Any, value: Any) -> None:
     # just accepting any dict.
     ok(MyTypedDictNotTotal, {"foo": "bar"})
     ok(MyTypedDict, {"foo": "bar", "blah": "meh"})
-    # Note, dicts can't have int key in JSON
-    fail(Dict[int, str], {1: "2"})
+
+    # Non-string dict keys are supported
+    ok(Dict[int, str], {1: "1"})
+    ok(Dict[float, str], {1.0: "1"})
+    ok(Dict[bool, str], {True: "1"})
+    ok(Dict[None, str], {None: "1"})
```

Comment:
Can you confirm that before this change, _all_ of these failed? I just want to make sure one of these didn't accidentally work _before_ we made these changes

Reply:
Yes I actually confirmed that same thing myself before marking the PR ready for review. They do all fail.